ls1intum/Apollon
# Support Syntax Trees

### Checklist

### Motivation and Context
Support simple syntax trees, used for the TUM lectures EidI and PGdP.

### Description
*Note: syntax tree is not a UML diagram, therefore I have not prefixed anything with UML.*
- Added the syntax-tree package
- Added all parts of the syntax-tree package to the components, elements, relationships, previews, diagram types and popups.
- Added localization for the syntax tree components

The preview/template elements are so small on purpose because the students will need lots of them and the text inside is usually very short.

### Steps for Testing
- Try out the new syntax tree diagram type, create terminals and nonterminals and link them with straight lines. It should look like shown in the screenshot below.
- Check that I haven't forgotten something and everything is correct TS code using the correct style.

### Screenshots
![image](https://user-images.githubusercontent.com/11130248/91206413-65631100-e707-11ea-99de-a0e1ac36209a.png)
